src/HTTPClient.ts

```diff
@@ -27,10 +27,41 @@ export class HTTPClient {
 	 * @public
 	 * @constructor
 	 * Creates an instance of a HTTPClient
-	 * @param {Middleware} middleware - The first middleware of the middleware chain
+	 * @param {Middleware| Middleware[]} middleware - The first middleware of the middleware chain or the array of middleware handlers
 	 */
-	public constructor(middleware: Middleware) {
-		this.middleware = middleware;
+	public constructor(middleware: Middleware | Middleware[]) {
+		this.setMiddleware(middleware);
+	}
+
+	/**
+	 * @private
+	 * Verifies if the middleware passed is an array or not before setting this.middleware property
+	 * @param {any} middleware - The middleware passed
+	 * @returns Nothing
+	 */
+
+	private setMiddleware(middleware: any): void {
+		if (Object.prototype.toString.call(middleware) === "[object Array]") {
+			this.parseMiddleWareArray(middleware);
+		} else {
+			this.middleware = middleware;
```

Comment:
```suggestion
			this.middleware = middleware[0];
```

Reply:
> The middleware is the else condition is not an array.

Reply:
@nikithauc this suggestion was to consider with the other ones, if you switch to var args it effectively becomes an array and simplifies your code

Reply:
and provides a strongly type api surface that supports both scenarios (one or multiple middlewares)

Reply:
@MIchaelMainer it will be if you switch to varargs :)

Reply:
Ugh, my comment came after you two had the conversation. Wish GH would update the conversation without refresh